### ISSUE TITLE
Porter bundles vs claims

### ISSUE THREAD

**Issue by vdice** [deislabs/porter#330]
A `porter bundle list` command was added in https://github.com/deislabs/porter/pull/329 which lists _installed_ bundles.  Technically, it is just listing the [claims](https://github.com/deislabs/cnab-spec/blob/master/400-claims.md) that have been generated/updated via `porter {install,upgrade,uninstall}` commands.  Therefore, the command would arguably be better categorized via `porter claim list`... and then, `porter claim show` could output a specific claim, etc.

Contrast this with the accepted interpretation of a `bundle`, which is a representation of a CNAB bundle, e.g., a [bundle.json file](https://github.com/deislabs/cnab-spec/blob/master/101-bundle-json.md).  One could/should feasibly be able to see a listing of bundles, if accessible (via local store, etc.), outside of needing to install them.  I believe Porter's current direction for listing these bundles is in https://github.com/deislabs/porter/issues/322.

In other words, a `bundle` is a representation of a CNAB bundle that may or may not have been installed and a `claim` represents an installation of a `bundle` in varying states.

Thoughts on how to more accurately portray a listing of bundle installations (claims) in Porter?  Simply update to `porter claim list`, opt for a different noun or other?

**Comment by squillace** [comment]
I propose either `installation` or `instance` for the active word used with created CNAB claims. It seems to be sitting well at the moment. HOWEVER, there's another design issue: Because I can push and pull to multiple machines but `installations` are local, I have discovered when moving machines that I cannot "remove" an "installation" in which the actual deployed artifacts are now gone.

What should the strategy be here? Should a `--local-destruction` option be available? The idea being that, if you REALLY know the resources aren't there, you can clean your local installations?

**Comment by carolynvs** [comment]
My vote is for `instance`

```
$ porter bundle instances list

# aliased to

$ porter list
```

I had two thoughts about bundles that have been installed, and now you can't uninstall them because the invocation image isn't completing properly. One reason could be because the resource is already deleted.

1. We should be pushing bundle authors (and mixin authors) HARD to write bundles that are repeatable and focus on desired state more than imperative code. If a resource that the bundle manages is already gone, mixins that we write should gracefully handle that and return successfully. Documentation that we write should push authors down the same path.
2. `porter uninstall [INSTANCE] --force` (alternatives: `--abandon`, `--cleanup`). The flag would indicate to porter that it should run the bundle but regardless of success/failure it should remove the instance from its records at the end. We may want a `-y` with an interactive prompt when `-y` isn't provided explaining that this may cause resources managed by the bundle to not be removed, and that they may continue to be charged for them.

**Comment by carolynvs** [comment]
OK. Instances has won. I've slapped a 🚨 🍩 on it and it's up for grabs.
